### ISSUE TITLE
libopenmpt0-shlibs-0.5.10-1

### ISSUE THREAD

**Issue by dak180** [fink/fink-distributions#851]
```
The following package will be installed or updated:
 libopenmpt0-shlibs
Setting runtime build-lock...
dpkg-deb -b /sw/src/fink.build/root-fink-buildlock-libopenmpt0-shlibs-0.5.10-1 /sw/src/fink.build
dpkg-deb: building package `fink-buildlock-libopenmpt0-shlibs-0.5.10-1' in `/sw/src/fink.build/fink-buildlock-libopenmpt0-shlibs-0.5.10-1_2021.08.27-15.39.49_darwin-x86_64.deb'.
Installing build-lock package...
/sw/bin/dpkg-lockwait -i /sw/src/fink.build/fink-buildlock-libopenmpt0-shlibs-0.5.10-1_2021.08.27-15.39.49_darwin-x86_64.deb
Selecting previously deselected package fink-buildlock-libopenmpt0-shlibs-0.5.10-1.
(Reading database ... 190878 files and directories currently installed.)
Unpacking fink-buildlock-libopenmpt0-shlibs-0.5.10-1 (from .../fink-buildlock-libopenmpt0-shlibs-0.5.10-1_2021.08.27-15.39.49_darwin-x86_64.deb) ...
Setting up fink-buildlock-libopenmpt0-shlibs-0.5.10-1 (2021.08.27-15.39.49) ...

sudo -u fink-bld [ENV] sh -c /tmp/fink.MJAqh
env LANG=C LC_ALL=C /sw/bin/tar  --no-same-owner --no-same-permissions -xf /sw/src/libopenmpt-0.5.10+release.autotools.tar.gz
sudo -u fink-bld [ENV] sh -c /tmp/fink.gUjGK
/bin/cp /sw/src/zlib.pc .
sudo -u fink-bld [ENV] sh -c /tmp/fink.nwmsM
perl -pi.bak -e 's|10\.\[012\]\*|10.[012][,.]*|g' configure
sudo -u fink-bld [ENV] sh -c /tmp/fink.6RhGg
/tmp/fink.eIRa2
#!/bin/sh -ev
	export PKG_CONFIG_PATH=`pwd`/../pkgconfig
pwd
	./configure --prefix=/sw 	--enable-dependency-tracking \
	--disable-static \
	--without-pulseaudio \
	--without-portaudiocpp \
	--with-sdl2=no
checking for a BSD-compatible install... /sw/bin/ginstall -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /sw/bin/gmkdir -p
checking for gawk... no
checking for mawk... no
checking for nawk... no
checking for awk... awk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether make supports the include directive... yes (GNU style)
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... gcc3
checking for ar... ar
checking the archiver (ar) interface... ar
checking build system type... x86_64-apple-darwin16.7.0
checking host system type... x86_64-apple-darwin16.7.0
checking how to print strings... printf
checking for a sed that does not truncate output... /usr/bin/sed
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for fgrep... /usr/bin/grep -F
checking for ld used by gcc... /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld
checking if the linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) is GNU ld... no
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 196608
checking how to convert x86_64-apple-darwin16.7.0 file names to x86_64-apple-darwin16.7.0 format... func_convert_file_noop
checking how to convert x86_64-apple-darwin16.7.0 file names to toolchain format... func_convert_file_noop
checking for /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for archiver @FILE support... no
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from gcc object... ok
checking for sysroot... no
checking for a working dd... /bin/dd
checking how to truncate binary pipes... /bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking for dsymutil... dsymutil
checking for nmedit... nmedit
checking for lipo... lipo
checking for otool... otool
checking for otool64... no
checking for -single_module linker flag... yes
checking for -exported_symbols_list linker flag... yes
checking for -force_load linker flag... yes
checking how to run the C preprocessor... gcc -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... yes
checking for gcc option to produce PIC... -fno-common -DPIC
checking if gcc PIC flag -fno-common -DPIC works... yes
checking if gcc static flag -static works... no
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin16.7.0 dyld
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS value needed for large files... no
checking for pkg-config... /sw/bin/pkg-config
checking pkg-config is at least version 0.24... yes
checking for gcc... (cached) gcc
checking whether we are using the GNU C compiler... (cached) yes
checking whether gcc accepts -g... (cached) yes
checking for gcc option to accept ISO C89... (cached) none needed
checking whether gcc understands -c and -o together... (cached) yes
checking dependency style of gcc... (cached) gcc3
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking dependency style of g++... gcc3
checking how to run the C++ preprocessor... g++ -E
checking for ld used by g++... /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld
checking if the linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) is GNU ld... no
checking whether the g++ linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) supports shared libraries... yes
checking for g++ option to produce PIC... -fno-common -DPIC
checking if g++ PIC flag -fno-common -DPIC works... yes
checking if g++ static flag -static works... no
checking if g++ supports -c -o file.o... yes
checking if g++ supports -c -o file.o... (cached) yes
checking whether the g++ linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin16.7.0 dyld
checking how to hardcode library paths into programs... immediate
checking for ZLIB... yes
checking for MPG123... yes
checking for OGG... yes
checking for VORBIS... yes
checking for VORBISFILE... yes
checking for PORTAUDIO... yes
checking for SNDFILE... yes
checking for FLAC... yes
checking for gcc option to accept ISO C99... none needed
checking for gcc option to accept ISO Standard C... (cached) none needed
checking whether g++ supports C++17 features with -std=c++17... no
checking whether g++ supports C++17 features with +std=c++17... no
checking whether g++ supports C++17 features with -h std=c++17... no
checking whether g++ supports C++17 features with -std=c++1z... yes
checking whether C compiler accepts -fvisibility=hidden... yes
checking CFLAGS for maximum warnings... -Wall
checking whether C++ compiler accepts -fvisibility=hidden... yes
checking CXXFLAGS for maximum warnings... -Wall
checking for doxygen... /sw/bin/doxygen
checking for perl... /usr/bin/perl
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating libopenmpt/libopenmpt.pc
config.status: creating Doxyfile
config.status: creating config.h
config.status: executing depfiles commands
config.status: executing libtool commands
/usr/bin/make
/Applications/Xcode.app/Contents/Developer/usr/bin/make  all-am
g++ -std=c++1z -DHAVE_CONFIG_H -I.   -I/sw/include   -I/sw/include -I/sw/include -I/sw/include  -g -O2 -fvisibility=hidden -Wall -MT openmpt123/bin_openmpt123-openmpt123.o -MD -MP -MF openmpt123/.deps/bin_openmpt123-openmpt123.Tpo -c -o openmpt123/bin_openmpt123-openmpt123.o `test -f 'openmpt123/openmpt123.cpp' || echo './'`openmpt123/openmpt123.cpp
In file included from openmpt123/openmpt123.cpp:92:
./libopenmpt/libopenmpt.hpp:309:75: error: no type named 'byte' in namespace
      'std'
  ...int probe_file_header( std::uint64_t flags, const std::byte * data, std:...
                                                       ~~~~~^
./libopenmpt/libopenmpt.hpp:342:75: error: no type named 'byte' in namespace
      'std'
  ...int probe_file_header( std::uint64_t flags, const std::byte * data, std:...
                                                       ~~~~~^
./libopenmpt/libopenmpt.hpp:471:33: error: no member named 'byte' in namespace
      'std'
        module( const std::vector<std::byte> & data, std::ostream & log ...
                                  ~~~~~^
./libopenmpt/libopenmpt.hpp:482:21: error: no type named 'byte' in namespace
      'std'
        module( const std::byte * beg, const std::byte * end, std::ostre...
                      ~~~~~^
./libopenmpt/libopenmpt.hpp:482:44: error: no type named 'byte' in namespace
      'std'
        module( const std::byte * beg, const std::byte * end, std::ostre...
                                             ~~~~~^
./libopenmpt/libopenmpt.hpp:493:21: error: no type named 'byte' in namespace
      'std'
        module( const std::byte * data, std::size_t size, std::ostream ...
                      ~~~~~^
6 errors generated.
make[1]: *** [openmpt123/bin_openmpt123-openmpt123.o] Error 1
make: *** [all] Error 2
### execution of /tmp/fink.eIRa2 failed, exit code 2
### execution of /tmp/fink.6RhGg failed, exit code 2
2021-08-27 15:40:35.222 perl5.18[82950:36723514] GrowlApplicationBridge: Could not send open-document event to register this application with Growl because AESend returned -600/no eligible process with specified descriptor
Removing runtime build-lock...
Removing build-lock package...
/sw/bin/dpkg-lockwait -r fink-buildlock-libopenmpt0-shlibs-0.5.10-1
(Reading database ... 190879 files and directories currently installed.)
Removing fink-buildlock-libopenmpt0-shlibs-0.5.10-1 ...
Failed: phase compiling: libopenmpt0-shlibs-0.5.10-1 failed

Before reporting any errors, please run "fink selfupdate" and try again.

If you continue to have issues, please check to see if the FAQ on Fink's 
website solves the problem.  If not, ask on one (not both, please) of
these mailing lists:

	The Fink Users List <fink-users@lists.sourceforge.net>
	The Fink Beginners List <fink-beginners@lists.sourceforge.net>,

with a carbon copy to the maintainer:

	Hanspeter Niederstrasser <nieder@users.sourceforge.net>

Note that this is preferable to emailing just the maintainer directly,
since most fink package maintainers do not have access to all possible
hardware and software configurations.

Please try to include the complete error message in your report.  This
generally consists of a compiler line starting with e.g. "gcc" or "g++"
followed by the actual error output from the compiler.

Also include the following system information:
Package manager version: 0.45.3
Distribution version: selfupdate-rsync Fri Aug 27 15:39:28 2021, 10.12, x86_64
Trees: local/main remote/dak180 stable/main
Xcode.app: 9.2
Xcode command-line tools: 9.2.0.0.1.1510905681
Max. Fink build jobs:  1
```

**Comment by nieder** [comment]
Looks like Xcode9 has problems with c++17. This is what I get on 10.14.6/Xcode11.3.1
```
checking whether g++ supports C++17 features with -std=c++17... yes
checking whether C compiler accepts -fvisibility=hidden... yes
```
I don't know if there's a way to patch it. Ffmpeg is the only package that uses it, and it was fine with the earlier version, so it might need to be forked to be available at the earlier release for 10.12 and lower.

**Comment by dak180** [comment]
> Ffmpeg is the only package that uses it, and it was fine with the earlier version, so it might need to be forked to be available at the earlier release for 10.12 and lower.

Sounds like a plan to me.

**Comment by dak180** [comment]
```
checking whether g++ supports C++17 features with -std=c++17... no
checking whether g++ supports C++17 features with +std=c++17... no
checking whether g++ supports C++17 features with -h std=c++17... no
checking whether g++ supports C++17 features with -std=c++1z... yes
checking whether C compiler accepts -fvisibility=hidden... yes
```
It might work on 10.12 with a different switch.

**Comment by dak180** [comment]
According to emscripten-core/emscripten#6580 this requires libc++ 6.0. Not sure if that is actually on 10.12 with xcode 9.2 or not.

**Comment by dak180** [comment]
> Fixed by commit [dd4f83a](https://github.com/fink/fink-distributions/commit/dd4f83a534ef4be9f67ab83b3ffab443455a8ff8)

It also fails on 10.12; not sure what version it starts working.
